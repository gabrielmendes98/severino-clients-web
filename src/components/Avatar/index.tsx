import MuiAvatar, { AvatarProps } from '@mui/material/Avatar';

const Avatar = ({ children, ...props }: AvatarProps) => (
  <MuiAvatar {...props}>{children}</MuiAvatar>
);

export default Avatar;
