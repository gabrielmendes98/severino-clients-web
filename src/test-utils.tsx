import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
