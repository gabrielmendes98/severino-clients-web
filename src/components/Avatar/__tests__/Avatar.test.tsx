import { render } from 'test-utils';
import Avatar from '../index';

it('should match snapshot', () => {
  const { container } = render(
    <Avatar src="mock" alt="test">
      test
    </Avatar>,
  );
  expect(container).toMatchSnapshot();
});
