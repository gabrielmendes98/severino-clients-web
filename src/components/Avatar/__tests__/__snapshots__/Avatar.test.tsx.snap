// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="MuiAvatar-root MuiAvatar-circular css-61edrn-MuiAvatar-root"
  >
    <img
      alt="test"
      class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
      src="mock"
    />
  </div>
</div>
`;
